layouts/partials/textprocessing.html

```diff
@@ -35,23 +35,26 @@
       {{$inner := . | strings.TrimPrefix "[[" | strings.TrimSuffix "]]" }}
       <!-- split from alias -->
       {{$split := split $inner "|"}}
+
       <!-- separate link path -->
       {{$path := index $split 0}}
       {{$reference := split $path "#"}}
       <!-- path with heading link (i.e. $block) removed -->
       {{$title := index $reference 0}}
-      <!-- heading link -->
-      {{$block := default "" (index $reference 1)}}
-      {{$block = strings.TrimRight "/" (cond (eq $block "") $block (printf "#%s" $block)) | urlize | lower}}
-      {{$href := strings.TrimRight "/" ($page.GetPage $title).RelPermalink}}
+
       <!-- if alias given, use alias, else title -->
       {{$display := default $title (index $split 1)}}
-      <!-- ADDED LINE TO REMOVE SUBFOLDER FROM TITLE: -->
+      <!-- REMOVE SUBFOLDER FROM TITLE -->
       {{$display := index (last 1 (split $display "/")) 0}}
-      {{if not $href}}
-        {{$link := printf "<a class=\"internal-link broken\">%s</a>" $display}}
+
+      {{if eq (string ($page.GetPage $title)) "nopPage"}}
```

Comment:
Good point. I guess the case will happen after this check for whether the link is a Hugo page.

Maybe if it's not a Hugo page and `$path` does not exist, the link should be broken? Let me know if you already know how to check if `$path` exists.

Reply:
we do it above! `$page.GetPage $title`